tuning/sft_trainer.py

```diff
@@ -353,16 +431,41 @@ def main(**kwargs):  # pylint: disable=unused-argument
     combined_tracker_configs.file_logger_config = file_logger_config
     combined_tracker_configs.aim_config = aim_config
 
-    train(
-        model_args=model_args,
-        data_args=data_args,
-        train_args=training_args,
-        peft_config=tune_config,
-        trainer_controller_args=trainer_controller_args,
-        tracker_configs=combined_tracker_configs,
-        additional_callbacks=None,
-        exp_metadata=metadata,
-    )
+    try:
+        train(
+            model_args=model_args,
+            data_args=data_args,
+            train_args=training_args,
+            peft_config=tune_config,
+            trainer_controller_args=trainer_controller_args,
+            tracker_configs=combined_tracker_configs,
+            additional_callbacks=None,
+            exp_metadata=metadata,
+        )
+    except (MemoryError, OutOfMemoryError) as e:
+        logger.error(traceback.format_exc())
+        write_termination_log(f"OOM error during training. {e}")
+        sys.exit(INTERNAL_ERROR_EXIT_CODE)
+    except FileNotFoundError as e:
+        logger.error(traceback.format_exc())
+        write_termination_log("Unable to load file: {}".format(e))
+        sys.exit(USER_ERROR_EXIT_CODE)
+    except HFValidationError as e:
```

Comment:
Note that in order to crop up these errors from SFT trainer, this error catching code needs to exist after the accelerate launch call. Otherwise the accelerate launch call only produces subprocess errors that hide the actual error that was hit. Thus moves from launch_training to sft_trainer but this means that anyone using sft_trainer or the library now will get an `error.log` file created and it's exiting with USER_ERROR_EXIT_CODE or INTERNAL_ERROR_EXIT_CODE instead of the actual python error code.